### ISSUE TITLE
Support `deno.jsonc` for Deno config file

### ISSUE THREAD

**Issue by binyamin** [lumeland/lume#211]
### Summary
Currently, Lume only supports `deno.json` files. Deno itself supports `deno.jsonc` files as well. Ideally, Lume would match Deno's behavior.

### Brainstorming
- Deno's standard library includes a JSONC parser in [`std/encoding`](https://deno.land/std@0.147.0/encoding/jsonc.ts). It does _not_ include a method for _serializing_ JSONC (see https://github.com/denoland/deno_std/discussions/1803).
- We only write to the deno config after the _vendor_ and _init_ commands. That's not too often. It might be worth parsing the JSONC using `std/encoding`, and doing some RegExp magic to overwrite a portion of the file. Just brainstorming here.

### Notes
- This feature is not in any way critical. I simply wanted to track JSONC support, and to explain why Lume does not in fact support JSONC.

**Comment by oscarotero** [comment]
As you said, Lume needs to read and write `deno.json` files to configure tasks and the import map file. In addition to `vendor` and `init`, this file is updated after `upgrade` and `import-map` commands.

Personally, I think it was a bad idea from Deno developers to allow to use jsonc here, it's not standard, introduce a unneeded complexity and inconsistency (what happens if `deno.json` and `deno.jsonc` exists in the same folder?, why it can be used for deno config but not for import_map.jsonc?), it's hard to parse and, specially to write while keeping the existing comments.

When Deno standard library include a option to parse and stringify JSONC, I'll consider support it, althought I think everybody should use json format.

**Comment by kidonng** [comment]
> why it can be used for deno config but not for import_map.jsonc?

Because Deno follows the spec, which only uses JSON: https://github.com/denoland/deno/issues/12993

And that leads me to another rabbit hole: you can specify import maps in `deno.json`! I don't think this is widely used, but 🤷‍♂️

**Comment by ooker777** [comment]
There is a std lib for jsonc now: [@std/jsonc - JSR](https://jsr.io/@std/jsonc "@std/jsonc - JSR")

**Comment by oscarotero** [comment]
@ooker777 You're right. In recent Lume versions the upgrade script has changed and jsonc files are not loaded. I'll fix it.
